contrib/keyboard-layout-per-window.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+# This script keeps track of active keyboard layouts per window.
+#
+# This script requires i3ipc-python package (install it from a system package
+# manager or pip).
+
+import i3ipc
+
+ipc = i3ipc.Connection()
+prev_focused = -1
+windows = {}
+
+def on_window_focus(ipc, event):
+    global windows, prev_focused
+
+    # Save current layout
+    layouts = {}
+    for input in ipc.get_inputs():
+        layouts[input.identifier] = input.xkb_active_layout_index
+    windows[prev_focused] = layouts
+
+    # Restore layout of the newly focused window
+    if event.container.id in windows:
+        for (input_id, layout_index) in windows[event.container.id].items():
```

Comment:
I think `input_id` may be better named as `kbd_id`, or perhaps `input_dev_id` *(I don't know can the device be anything else besides a keyboard?)*.

Which btw raises a question: what happens if we detached a keyboard? I guess the dictionary now gonna have stale entries, that needs to be cleaned up.

Reply:
>I think input_id may be better named as kbd_id, or perhaps input_dev_id (I don't know can the device be anything else besides a keyboard?).

-1

Reply:
> -1

-1, because… ?

> This could even be seen as a feature: if you re-plug the keyboard, you don't loose your window → layout mappings.

Fair enough, but are you sure this ID is always the same for the same keyboard?

Reply:
@emersion please un-resolve this discussion. For some reason I can't do this, and this one is ongoing.